pkg/kubelet/cm/dra/manager.go

```diff
@@ -108,30 +109,25 @@ func (m *ManagerImpl) PrepareResources(pod *v1.Pod) error {
 			continue
 		}
 
-		claimInfo := m.cache.get(*claimName, pod.Namespace)
-		if claimInfo == nil {
-			// claim does not exist in cache, create new claimInfo object
-			// to be processed later.
-			claimInfo = newClaimInfoFromResourceClaim(resourceClaim)
+		if m.cache.contains(*claimName, pod.Namespace) {
+			// Claim exists in cache. Reference the pod.
+			err = m.cache.addPodReference(*claimName, pod.Namespace, pod.UID)
+			if err != nil {
+				return fmt.Errorf("claim %s: failed to reference pod UID %s: %w", *claimName, pod.UID, err)
+			}
 		}
```

Comment:
There's a race with the deletion of the cache entry here. I guess that's OK and unavoidable? After failing, the next try will *not* have the claim in the cache and proceed to go down the path of preparing it again.

Reply:
Either way, I don't like this and still think we can do something simpler and more readable.

Reply:
Yes, that's also my understanding.